### PR TITLE
Add Bitbucket Pipelines variable module

### PR DESCRIPTION
**Pull request by catcombo**
##### SUMMARY

Manages [Bitbucket](http://bitbucket.org/) Pipelines variables.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

bitbucket_pipelines_variable

### CODE REVIEW

**Comment by zyv** [comment]
@Akasurde could you please be so kind to help us to find reviewers for this new module? We tried our best to produce a clean one according to the community guidelines and it also has full test coverage to ease reviews. Thank you!

Besides, we are aiming to open source a few more Bitbucket modules, but they all build upon this PR. What would be the best way to proceed? Wait until this one is through and open more PRs, one PR per new module?

/cc @dagwieers hi there, here I come :-) 🍻

File: lib/ansible/module_utils/bitbucket.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
```
Comment (dagwieers) [inline lib/ansible/module_utils/bitbucket.py]:
```suggestion
# -*- coding: utf-8 -*-

# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/python
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
#!/usr/bin/python
# -*- coding: utf-8 -*-
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community',
+}
+
+DOCUMENTATION = '''
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
DOCUMENTATION = r'''
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+short_description: Manages Bitbucket pipelines variables
+description:
+  - Manages Bitbucket pipelines variables.
+version_added: "2.8"
+author:
+  - Evgeniy Krysanov (@catcombo)
+options:
+  client_id:
+    description:
+      - OAuth consumer key. If not set the environment variable C(BITBUCKET_CLIENT_ID) will be used.
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
      - OAuth consumer key. If not set the environment variable C(BITBUCKET_CLIENT_ID) will be used.
    type: str
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+  client_secret:
+    description:
+      - OAuth consumer secret. If not set the environment variable C(BITBUCKET_CLIENT_SECRET) will be used.
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
      - OAuth consumer secret. If not set the environment variable C(BITBUCKET_CLIENT_SECRET) will be used.
    type: str
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+  repository:
+    description:
+      - Repository name.
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
      - Repository name.
    type: str
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+    required: true
+  username:
+    description:
+      - Repository owner.
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
      - Repository owner.
    type: str
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+    required: true
+  key:
+    description:
+      - Pipeline variable name.
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
      - Pipeline variable name.
    type: str
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+    required: true
+  value:
+    description:
+      - Pipeline variable value.
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
      - Pipeline variable value.
    type: str
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+  secured:
+    description:
+      - Encrypt variable value.
+    default: no
+    type: bool
+  state:
+    description:
+      - Indicates desired state of the variable.
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
      - Indicates desired state of the variable.
    type: str
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+    choices:
+      - present
+      - absent
+notes:
+  - Bitbucket OAuth consumer key and secret can be obtained from Bitbucket profile -> Settings -> Access Management -> OAuth.
+  - Check mode is supported.
+  - For secured values return parameter C(changed) is always C(True).
+'''
+
+EXAMPLES = '''
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
EXAMPLES = r'''
```

File: lib/ansible/modules/source_control/bitbucket_pipelines_variable.py [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+
+- name: Remove pipeline variable
+  bitbucket_pipelines_variable:
+    repository: bitbucket-repo
+    username: bitbucket_username
+    key: AWS_ACCESS_KEY
+    state: absent
+'''
+
+RETURN = ''' # '''
```
Comment (dagwieers) [inline lib/ansible/modules/source_control/bitbucket_pipelines_variable.py]:
```suggestion
RETURN = r''' # '''
```

**Review by dagwieers — CHANGES_REQUESTED** [review]
Looks good. A few cosmetic changes only. I did not test the module.

**Comment by dagwieers** [comment]
@zyv Yes, it's best to get the module_utils in first, and then add the rest.
Also, the community freeze is next week, which does not give a lot of time to get these included in Ansible 2.8.

Going forward, it could help to tap into the existing Bitbucket community to find people that need/use these modules and form a Working Group to increase the testing "clout".

**Comment by catcombo** [comment]
@ansibot ready_for_review

**Review by dagwieers — APPROVED** [review]

**Review by zyv — APPROVED** [review]
Awesome 👍

**Review by dagwieers — APPROVED** [review]
LGTM, thanks !

**Comment by dagwieers** [comment]
@marns93 You can now create PRs for the other modules. Please take into account all remarks we made for this module. Thanks!

**Comment by zyv** [comment]
@dagwieers oh wow, thanks! 🎉 

Just to set the record straight ;), the maintainers will be @zyv (me), @marns93 and @catcombo. We'll take into account your comments for this module, clean up other modules to match style, and make new PRs, hopefully today still.

**Pull request merged**